app/src/main/java/app/revanced/integrations/youtube/patches/misc/HookDownloadAction.java

```diff
@@ -0,0 +1,28 @@
+package app.revanced.integrations.youtube.patches.misc;
+
+import app.revanced.integrations.youtube.settings.Settings;
+import app.revanced.integrations.youtube.utils.VideoUtils;
+
+@SuppressWarnings("unused")
+public class HookDownloadAction {
+
+    /*
+     * Injection point
+     */
+    public static boolean isPlaylistDownloadButtonHooked() {
+        return Settings.HOOK_PLAYLIST_DOWNLOAD_BUTTON.get();
+    }
+
+    /*
+     * Injection point
+     */
+    public static String startPlaylistDownloadActivity(String playlistId) {
```

Comment:
Hooking is done before checking if playlistId is null or empty.
In integrations, you need to check if playlistId is null or empty.